jormungandr/src/explorer/graphql/mod.rs

```diff
@@ -63,16 +64,56 @@ impl Block {
     }
 
     /// The transactions contained in the block
-    pub fn transactions(&self, context: &Context) -> FieldResult<Vec<Transaction>> {
-        Ok(self
-            .get_explorer_block(&context.db)?
-            .transactions
-            .iter()
-            .map(|(id, _tx)| Transaction {
-                id: id.clone(),
-                in_block: self.hash.clone(),
-            })
-            .collect())
+    pub fn transactions(
+        &self,
+        first: Option<i32>,
+        last: Option<i32>,
+        before: Option<IndexCursor>,
+        after: Option<IndexCursor>,
+        context: &Context,
+    ) -> FieldResult<TransactionConnection> {
+        let explorer_block = self.get_explorer_block(&context.db)?;
+        let mut transactions: Vec<&ExplorerTransaction> =
+            explorer_block.transactions.values().collect();
+
+        // TODO: This may be expensive at some point, but I can't rely in
+        // the HashMap's order (also, I'm assuming the order in the block matters)
+        transactions
+            .as_mut_slice()
+            .sort_unstable_by_key(|tx| tx.offset_in_block);
+
+        let lower_bound = 0u32;
+        let upper_bound = transactions.len();
+
+        let pagination_arguments = PaginationArguments {
+            first,
+            last,
+            before: before.map(u32::try_from).transpose()?,
+            after: after.map(u32::try_from).transpose()?,
+        }
+        .validate()?;
+
+        TransactionConnection::new(
+            lower_bound,
+            upper_bound
+                .try_into()
+                .expect("tried to paginate more than 2^32 elements"),
+            pagination_arguments,
+            |from: u32, to: u32| {
+                let from = usize::try_from(from).unwrap();
+                let to = usize::try_from(to).unwrap();
+
+                (from..to)
+                    .map(|i| {
+                        (
+                            transactions[i].id().clone(),
+                            i.try_into()
+                                .expect("tried to paginate more than 2^32 elements"),
```

Comment:
This is now infallible as `both` from and `to` are `u32`?

Reply:
yes, good eye